### PR TITLE
itk 5.4.0

### PR DESCRIPTION
**Pull request by thewtex**
- ENH: Update CI for ITK 5.4.0
- ENH: Migrate python packageing config to pyproject.toml
- ENH: Use setuptools_scm to set Python package version

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,110 @@
+[build-system]
+requires = ["scikit-build-core"]
+build-backend = "scikit_build_core.build"
+
+[project]
+name = "itk-cuberille"
+version = "2.6.0"
+description = "Convert an ITK Mesh to a simple data structure compatible with vtkPolyData."
```
Comment (dzenanz) [inline pyproject.toml:7]: This field needs an update.

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+[project.urls]
+Download = "https://github.com/InsightSoftwareConsortium/ITKCuberille"
+Homepage = "https://github.com/InsightSoftwareConsortium/ITKCuberille"
+
+[tool.scikit-build]
+# The versions of CMake to allow. If CMake is not present on the system or does
+# not pass this specifier, it will be downloaded via PyPI if possible. An empty
+# string will disable this check.
+cmake.version = ">=3.16"
```
Comment (dzenanz) [inline pyproject.toml:57]: 3.16.3 (is the patch version important?)

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+backport.find-python = "3.26.1"
+
+# Select the editable mode to use. Can be "redirect" (default) or "inplace".
+editable.mode = "redirect"
+
+# Rebuild the project when the package is imported. The build-directory must be
+# set.
+editable.rebuild = false
+
+# If set, this will provide a method for backward compatibility.
```
Comment (dzenanz) [inline pyproject.toml:112]: scikit backward compatibility. Preferably base this file on https://github.com/InsightSoftwareConsortium/ITKModuleTemplate/blob/main/%7B%7Bcookiecutter.project_name%7D%7D/pyproject.toml

File: pyproject.toml (hunk 2 above)
Comment (thewtex) [reply pyproject.toml:57]: Not really

**Review by dzenanz — APPROVED** [review]

**Pull request merged**